plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -247,6 +270,16 @@ func (c *X509Cert) Gather(acc telegraf.Accumulator) error {
 }
 
 func (c *X509Cert) Init() error {
+	err := c.refreshFilePaths()
+	if err != nil {
+		return err
+	}
+
+	err = c.sourcesToURLs()
+	if err != nil {
+		return err
+	}
+
```

Comment:
I think it's better to swap these functions so `refreshFilePaths()` (which might be called `expandFilePaths()`)  can work on normalized inputs as otherwise you miss entries where the user specified files in the form `file:///etc/certs/*`.